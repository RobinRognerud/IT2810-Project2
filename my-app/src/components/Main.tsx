import React, { useState } from "react";
import Music from "./MusicComp/Music";
import MusicOpt from "./MusicComp/MusicOpt";
import DisplayPicture from "./PictureComp/DisplayPicture";
import PictureOpt from "./PictureComp/PictureOpt";
import Poetry from "./PoetryComp/Poetry";
import PoetryOpt from "./PoetryComp/PoetryOpt";

import "../App.css";
import SaveFavoriteButton from "./FavoriteButton/SaveFavoriteButton";
import GetFavoriteButton from "./FavoriteButton/GetFavoriteButton";

interface IMainProps {}

const Main: React.FC<IMainProps> = (props) => {
  /*Initialize radiobutton values to 0 if nothing is saved in localstorage*/
  const [pictureNo, setPictureNo] = useState(
    sessionStorage.getItem("SessionStoragePictureNo") || "0"
  );

  const [musicNo, setMusicNo] = useState(
    sessionStorage.getItem("SessionStorageMusicNo") || "0"
  );

  const [poetryNo, setPoetryNo] = useState(
    sessionStorage.getItem("SessionStoragePoetryNo") || "0"
  );

  /* Save the radiobutton values to localstorage after favoritebutton is clicked */
  const saveFavorite = () => {
    localStorage.setItem("LocalStoragePictureNo", pictureNo);
    localStorage.setItem("LocalStorageMusicNo", musicNo);
    localStorage.setItem("LocalStoragePoetryNo", poetryNo);
  };

  /* Update pictureNo after a radiobutton is clicked */
  const updatePictureNo = (e: React.ChangeEvent<HTMLInputElement>) => {
    sessionStorage.setItem("SessionStoragePictureNo", e.target.value);
    setPictureNo(e.target.value);
  };

  /* Update musicNo after a radiobutton is clicked */
  const updateMusicNo = (e: React.ChangeEvent<HTMLInputElement>) => {
    sessionStorage.setItem("SessionStorageMusicNo", e.target.value);
    setMusicNo(e.target.value);
  };

  /* Update poetryNo after a radiobutton is clicked */
  const updatePoetryNo = (e: React.ChangeEvent<HTMLInputElement>) => {
    sessionStorage.setItem("SessionStoragePoetryNo", e.target.value);
    setPoetryNo(e.target.value);
  };

  const getFavorite = () => {
    const pictureNo = localStorage.getItem("LocalStoragePictureNo") || "0";
    const musicNo = localStorage.getItem("LocalStorageMusicNo") || "0";
    const poetryNo = localStorage.getItem("LocalStoragePoetryNo") || "0";

    setPictureNo(pictureNo);
    setMusicNo(musicNo);
    setPoetryNo(poetryNo);
  };

  return (
    <div className="home-container" id="home">
      {/* <div className="gallery"> */}
      <div className="picture-container">
        <DisplayPicture pictureNo={pictureNo} />
      </div>
      <div className="text-container">
        <Poetry poetryNo={poetryNo} />
        <Music musicNo={musicNo} />
      </div>
      {/* </div> */}
      <div className="options-container">
        <p> Make your own art by changing image, music and poetry:</p>

        <PictureOpt updatePictureNo={updatePictureNo} pictureNo={pictureNo} />

        <MusicOpt updateMusicNo={updateMusicNo} musicNo={musicNo} />

        <PoetryOpt updatePoetryNo={updatePoetryNo} poetryNo={poetryNo} />

        <SaveFavoriteButton saveFavorite={saveFavorite} />
        <GetFavoriteButton getFavorite={getFavorite} />
      </div>
    </div>
  );
};

export default Main;
