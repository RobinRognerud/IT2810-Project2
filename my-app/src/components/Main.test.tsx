import React from "react";
import renderer from "react-test-renderer";

import Main from "./Main";

it("renders main correctly", () => {
  const tree = renderer.create(<Main />).toJSON();
  expect(tree).toMatchSnapshot();
});
