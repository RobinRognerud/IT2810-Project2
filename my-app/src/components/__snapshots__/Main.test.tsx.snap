// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders main correctly 1`] = `
<div
  className="home-container"
>
  <div
    className="picture-container"
  >
    Bilder
    <div
      id="product-options"
    >
      <div
        id="picture-category"
      >
        <img
          alt=""
          className="Svg"
          src="https://www.svgrepo.com/show/30310/sun.svg"
        />
      </div>
    </div>
  </div>
  <div
    className="text-container"
  >
    Dikt + lyd
    <div
      className="poetry"
    >
      <p>
        
      </p>
    </div>
    <div>
      <audio
        controls={true}
        src="Prologue.mp3"
      />
    </div>
  </div>
  <div
    className="options-container"
  >
    <div
      id="product-options"
    >
      <div
        id="picture-category"
      >
        <label>
          Choice 1:
          <input
            checked={true}
            name="pictureNr"
            onChange={[Function]}
            type="radio"
            value="0"
          />
        </label>
        <label>
          Choice 2:
          <input
            checked={false}
            name="pictureNr"
            onChange={[Function]}
            type="radio"
            value="1"
          />
        </label>
        <label>
          Choice 3:
          <input
            checked={false}
            name="pictureNr"
            onChange={[Function]}
            type="radio"
            value="2"
          />
        </label>
      </div>
    </div>
    <br />
    <br />
    <div
      id="product-options"
    >
      <div
        id="music-category"
      >
        <label>
          Choice 1:
          <input
            checked={true}
            name="musicNr"
            onChange={[Function]}
            type="radio"
            value="0"
          />
        </label>
        <label>
          Choice 2:
          <input
            checked={false}
            name="musicNr"
            onChange={[Function]}
            type="radio"
            value="1"
          />
        </label>
        <label>
          Choice 3:
          <input
            checked={false}
            name="musicNr"
            onChange={[Function]}
            type="radio"
            value="2"
          />
        </label>
      </div>
    </div>
    <br />
    <br />
    <div
      id="product-options"
    >
      <div
        id="poetry-category"
      >
        <label>
          Choice 1:
          <input
            checked={true}
            name="poetryNr"
            onChange={[Function]}
            type="radio"
            value="0"
          />
        </label>
        <label>
          Choice 2:
          <input
            checked={false}
            name="poetryNr"
            onChange={[Function]}
            type="radio"
            value="1"
          />
        </label>
        <label>
          Choice 3:
          <input
            checked={false}
            name="poetryNr"
            onChange={[Function]}
            type="radio"
            value="2"
          />
        </label>
      </div>
    </div>
    <br />
    <br />
    <div>
      <p>
        Lagre din favoritt utstillingen til neste gang!
      </p>
      <button
        onClick={[Function]}
      >
        Lagre utstilling
      </button>
    </div>
  </div>
</div>
`;
